### PR TITLE
Subscription debugging/monitoring

### PR DESCRIPTION
**Pull request by derekcollison**
Added code to allow subscription detail under /subsz with ?subs=1.

Also allow testing against a publish subject with /subsz?subs=1&test=foo. Will only return subscriptions that would match "foo".

Added ability to limit number of subscriptions per connection.

/cc @nats-io/core

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -766,6 +766,11 @@ func (s *Server) createClient(conn net.Conn) *client {
 		return c
 	}
 
+	// If there is a max subscriptions specified, add to the client.
```
Comment (kozlovic) [inline server/server.go]: Should this be moved down (or up) to under the client lock instead of server lock?

File: server/monitor.go [hunk 2]
```diff
@@ -638,9 +632,119 @@ func (s *Server) HandleRoutez(w http.ResponseWriter, r *http.Request) {
...
+		}
+		limit = opts.Limit
+		if limit <= 0 {
+			limit = DefaultSubListSize
+		}
+		if opts.Test != "" {
+			testSub = opts.Test
+			test = true
+			if !IsValidLiteralSubject(testSub) {
+				return nil, fmt.Errorf("Invalid test subject, must be valid publish subject:: %s", testSub)
```
Comment (kozlovic) [inline server/monitor.go]: Is `::` intentional here?

Comment (derekcollison) [reply server/monitor.go]: No ;)

File: server/server.go (hunk 1 above)
Comment (derekcollison) [reply server/server.go]: Will fix..

**Comment by kozlovic** [comment]
LGTM

**Pull request merged**